src/bluesky/run_engine.py

```diff
@@ -1965,13 +1972,15 @@ async def _declare_stream(self, msg):
             raise IllegalMessageSequence(ims_msg)
         return await current_run.declare_stream(msg)
 
-    async def _read(self, msg):
+    async def _read(self, msg: Msg):
```

Comment:
Is it worth refactoring `_read` to follow the same pattern you used in `_trigger` to share more code between the two versions?